hera_stats/jkset.py

```diff
@@ -0,0 +1,319 @@
+
+        self.dlys = dlys
+        self.shape = self._uvp_list.shape
+        self._validate()
+
+    def __getitem__(self, key):
+        """
+        Handles indexing.
+        """
+        # Let numpy handle the indexing
+        new_uvp = self._uvp_list[key]
+
+        # Make 2D if not already 2d.
+        if isinstance(new_uvp, hp.UVPSpec):
+            new_uvp = np.array([[new_uvp]])
+        if new_uvp.ndim == 1:
+            new_uvp = new_uvp[None]
+
+        # Create new JKSet with sliced uvp_list
+        newjk = JKSet(new_uvp, self.jktype)
+        return newjk
+
+    def __repr__(self):
+        """
+        Handles printing.
+        """
+        # Recreate normal class representation
+        name = "<hera_stats.jkset.JKSet instance at {}>".format(hex(id(self)))
+
+        # Add some informative information
+        info = ("Jackknife Data\n"
+                "--------------\n"
+                "jktype: {}\n"
+                "data shape: {}\n").format(self.jktype, self._uvp_list.shape)
+        return name + "\n\n" + info
+    
+    def __eq__(self, jk2, just_meta=False):
+        """
+        Handles ==
+        """
+        # Validate both data sets
+        self._validate(), jk2._validate()
+
+        # Attributes to check if equal (add actual data to list if not just_meta)
+        attrs = ["nsamples", "times", "units", "vis_units", "integrations", "grps", "jktype"]
+        if not just_meta:
+            attrs.extend(["spectra", "errs"])
+
+        # Check if equal
+        for attr in attrs:
+            if not np.all(getattr(self, attr) == getattr(jk2, attr)):
+                return False
+
+        return True
+
+    def add(self, jkset2, axis=1, inplace=False):
+        """
+        Stacks together the current jkset and another along a specific axis.
+
+        Parameters
+        ----------
+        jkset2: JKSet
+            The other JKSet to stack with this one.
+
+        axis: int, 1 or 0, optinoal
+            Axis along which to add. Default: 1.
+
+        inplace: boolean, optional
+            If true, keeps stacked data in this object. If false, returns new object. Default: False.
+
+        Returns
+        -------
+        new_jkset: JKSet
+            If inplace == False, returns a jkset with spectra of both this class and jkset2
+        """
+        # Test shape to see if stackable
+        assert axis in [0, 1], "axis must either be 1 or 0."
+        assert self.shape[not axis] == jkset2.shape[not axis], "jksets don't match shape in dimension {}.".format(not axis)
+
+        # Vstack or hstack uvp lists
+        if axis == 1:
+            new_uvp = np.hstack([self._uvp_list, jkset2._uvp_list])
+        elif axis == 0:
+            new_uvp = np.vstack([self._uvp_list, jkset2._uvp_list])
+
+        # Return or set data
+        if inplace:
+            self._load_uvp(new_uvp)
+        else:
+            return JKSet(new_uvp, self.jktype)
+
+    def set_data(self, spectra, errs):
```

Comment:
Should we have an `err_type` field to specify what type of errors these are? They might be taken from different fields in the `UVPSpec.stats_array`.